flask/tests/test_users.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+
+def test_list_users(test_database, client, login_as):
+    login_as("admin")
+    response = client.get("/api/users")
+    assert response.status_code == 200
+
+    user_list = response.get_json()
+    assert len(user_list) == 2
+
+    # return order is unspecified
+    user_list.sort(key=lambda user: user["username"])
```

Comment:
Clever! Out of curiosity, does SQLAlchemy by default obey some kind of ordering or are collections returned in no particular order.

Reply:
SQLAlchemy should be returning in the same order as the underlying SQL, which is unspecified unless an ORDER BY clause is used. So the default ordering is completely arbitrary and probably the order the database has in its index, which is likely insertion order.